lib/chef/resource/aws_cloudsearch_domain.rb

```diff
@@ -0,0 +1,46 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+module AWS
+  class CloudSearch
+    class Domain
+      # The version of the AWS sdk we are using doesn't have a model
```

Comment:
The `aws_cache_*` resources had this same problem.  For those we specified `aws_sdk_type AWS::ElastiCache`  We could do the same thing here unless we find an issue with it - `aws_sdk_type AWS::CloudSearch`